src/SFA.DAS.AssessorService.Application/Handlers/ExternalApi/DataSync/ExternalApiDataSyncHandler.cs

```diff
@@ -1,51 +1,48 @@
-﻿using Dapper;
-using SFA.DAS.AssessorService.ExternalApiDataSync.Infrastructure;
-using SFA.DAS.AssessorService.ExternalApiDataSync.Logger;
-using System;
+﻿using System;
 using System.Collections.Generic;
 using System.Data;
 using System.Data.SqlClient;
+using System.Threading;
 using System.Threading.Tasks;
 using System.Transactions;
+using Dapper;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.AssessorService.Settings;
 
-namespace SFA.DAS.AssessorService.ExternalApiDataSync
+namespace SFA.DAS.AssessorService.Application.Handlers.ExternalApi.DataSync
 {
-    public interface ICommand
+    public class ExternalApiDataSyncHandler : IRequestHandler<GetDataSyncRequest>
     {
-        Task Execute();
-    }
-
-    public class ExternalApiDataSyncCommand : ICommand
-    {
-        private readonly IAggregateLogger _aggregateLogger;
-        private readonly bool _allowDataSync;
-        private readonly string _sourceConnectionString;
-        private readonly string _destinationConnectionString;
+        private readonly ILogger<ExternalApiDataSyncHandler> _logger;
+        private readonly SqlBulkCopyOptions _bulkCopyOptions;
+        private readonly IWebConfiguration _config;
 
-        private readonly SqlBulkCopyOptions _bulkCopyOptions;     
-
-        public ExternalApiDataSyncCommand(IWebConfiguration config, IAggregateLogger aggregateLogger)
+        public ExternalApiDataSyncHandler(ILogger<ExternalApiDataSyncHandler> logger, IWebConfiguration configurationService)
         {
-            _aggregateLogger = aggregateLogger;
-
-            _allowDataSync = config.ExternalApiDataSync.IsEnabled;
-            _sourceConnectionString = config.SqlConnectionString;
-            _destinationConnectionString = config.SandboxSqlConnectionString;
+            _logger = logger;
+            _config = configurationService;
             _bulkCopyOptions = SqlBulkCopyOptions.KeepIdentity | SqlBulkCopyOptions.KeepNulls | SqlBulkCopyOptions.TableLock;
         }
 
-        public async Task Execute()
+        public async Task<Unit> Handle(GetDataSyncRequest request, CancellationToken cancellationToken)
```

Comment:
I think you should move the code again - this time into a repository - the connection strings are still to high up the stack